### PR TITLE
Add more flexible handling of metrics generation

### PR DESCRIPTION
**Pull request by CompaqDisc**
Also adds the option to not send invalid sensor data (default), as well as simplified 404 generation.

### CODE REVIEW

**Comment by kashalls** [comment]
LGTM 👍

File: AirGradient-Server/AirGradient-Server.ino [hunk 1]
```diff
@@ -91,70 +144,123 @@ void loop() {
...
+  TMP_RH stat = ag.periodicFetchData();
+  dtostrf(stat.t, 4, 2, atmp_reading);
+  return { atmp_reading, stat.error != 0 };
+}
+
+metric_resp_t get_rhum() {
+  // Early return if the sensor isn't enabled.
+  if (!hasSHT) return { "(disabled)", true };
+
+  TMP_RH stat = ag.periodicFetchData();
```
Comment (ssvenn) [inline AirGradient-Server/AirGradient-Server.ino:178]: Does this code actually work for you? This always returns 0 humidity on my airgradient DIY board, but the oled display shows the humidity value. I added some debug serial output and it seems like the first ag.periodicFetchData() call gets valid data in the get_atmp function but the second call immediately after in get_rhum returns a zero value whenever it is responding to a web request.

Comment (kashalls) [reply AirGradient-Server/AirGradient-Server.ino:178]: I'll look into it, I'm still waiting on my kit to arrive. It *was* working on my friends kit. I was waiting to write tests until I could verify it

Comment (ssvenn) [reply AirGradient-Server/AirGradient-Server.ino:178]: Great!
My ugly workaround that fixed it for me was to update the global variable during get_atmp like so:
metric_resp_t get_atmp() {
  // Early return if the sensor isn't enabled.
  if (!hasSHT) return { "(disabled)", true };

  TMP_RH stat = ag.periodicFetchData();
  dtostrf(stat.t, 4, 2, atmp_reading);
  itoa(stat.rh, rh_reading, 10);
  return { atmp_reading, stat.error != 0 };
}

metric_resp_t get_rhum() {
  // Early return if the sensor isn't enabled.
  if (!hasSHT) return { "(disabled)", true };

  return { rh_reading, true };
}

**Comment by kashalls** [comment]
@CompaqDisc can we verify that this is working on your kit?

File: AirGradient-Server/AirGradient-Server.ino (hunk 1 above)
Comment (kashalls) [reply AirGradient-Server/AirGradient-Server.ino:178]: > Great! My ugly workaround that fixed it for me was to update the global variable during get_atmp like so: metric_resp_t get_atmp() { // Early return if the sensor isn't enabled. if (!hasSHT) return { "(disabled)", true };
> 
> TMP_RH stat = ag.periodicFetchData(); dtostrf(stat.t, 4, 2, atmp_reading); itoa(stat.rh, rh_reading, 10); return { atmp_reading, stat.error != 0 }; }
> 
> metric_resp_t get_rhum() { // Early return if the sensor isn't enabled. if (!hasSHT) return { "(disabled)", true };
> 
> return { rh_reading, true }; }

Just got mine today, built it and ran into this too.

**Pull request merged**